examples/src/main/scala/com/cloudera/livy/apps/WordCountApp.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to Cloudera, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Cloudera, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cloudera.livy.apps
+
+import java.io.{File, FileNotFoundException}
+import java.net.URI
+import java.util.Properties
+
+import org.apache.spark.launcher.SparkLauncher
+import org.apache.spark.storage.StorageLevel
+import scala.concurrent.Await
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+import com.cloudera.livy.LivyClientBuilder
+import com.cloudera.livy.scalaapi.{LivyScalaClient, ScalaJobHandle, _}
+
+/**
+  * A WordCount example using Scala-API which reads text from a stream and saves
+  * it as data frames. The word with maximum count is the result
+  */
+object WordCountApp {
+
+  var scalaClient: LivyScalaClient = null
+
+  def init(url: String): Unit = {
+    val conf = new Properties
+    val classpath: String = System.getProperty("java.class.path")
+    conf.put("spark.app.name", "ScalaWordCount app")
+    conf.put(SparkLauncher.DRIVER_EXTRA_CLASSPATH, classpath)
+    conf.put(SparkLauncher.EXECUTOR_EXTRA_CLASSPATH, classpath)
+    scalaClient = new LivyClientBuilder(true).setURI(new URI(url)).build().asScalaClient
+  }
+
+  def uploadScalaAPIJar(): Unit = {
+    var appTargetFolderPath = getClass().getProtectionDomain.getCodeSource.getLocation.getPath
```

Comment:
This can return a path that is not a jar file, or may even return `null`? You should check for those things and print an appropriate message.